dlrover/python/master/shard_manager/dataset_splitter.py

```diff
@@ -289,3 +318,139 @@ def create_dataset_splitter(
+
+
+class StreamingDatasetSplitter(DatasetSplitter):
+    """StreamingDatasetSplitter split a dataset stored in a message queue like 
+    Kafka or SLS. We can read data by record offset in the logstore.
+    The shard contains index ranges [start, end) and partition of batch records.
+    Attributes:
+        dataset_name: the name of the logstore.
+        shuffle: whether to shuffle shards of the dataset, by default is False
+        batch_size: the number of records in a batch.
+        data_size: the number of records of the dataset 
+        max_shard_count: the max number of shards in the memory.
+            The value can limit the number of shards in the memory
+            to avoid OOM.
+    """
+
+    STORAGE_TYPE = "sls"
+    def __init__(
+        self,
+        dataset_name,
+        shard_size,
+        partition_offset,
+        num_epochs=1,
+        dataset_size=-1,
+        shuffle=False,
+        max_shard_count=None,
+        fetch_data_size=10000,
+        ):
+        super(StreamingDatasetSplitter, self).__init__(
+            dataset_name,
+            dataset_size,
+            shard_size,
+            num_epochs,
+        )
+        self._dataset_name = dataset_name
+        self._shard_size = shard_size
+        self._partition_offset = partition_offset
+        self.num_epochs = num_epochs
+        self._dataset_size = dataset_size
+        self._shuffle = shuffle
+        self._max_shard_count = max_shard_count
+        self._fetch_data_size = fetch_data_size
+        self._shards = []
+        self.epoch = 0
+ 
+    def update_partition_offsets(self, partition_offset):
+        self._partition_offset = partition_offset
+
+    def epoch_finished(self):
+        finished = False
+        if self._dataset_size == 0:
+            finished = True
+        return finished
+        
+    def get_epoch(self):
+        return 1 
+
+    def get_fetch_data_size(self):
+        return self._fetch_data_size
+
+    def to_checkpoint(self):
+        partition_offset = self._partition_offset.to_json()
+        checkpoint = self.__dict__
+        checkpoint.update({"_partition_offset":partition_offset})
+        return checkpoint
+
+    @staticmethod
+    def from_checkpoint(checkpoint):
+        init_args = {}
+        args_key = ["dataset_name",
+                    "shard_size",
+                    "partition_offset",
+                    "num_epochs",
+                    "dataset_size",
+                    "shuffle",
+                    "max_shard_count",
+                    "fetch_data_size"]
+        partition_offset = PartitionOffsets(checkpoint.pop("_partition_offset"))
+        for k in args_key:
+            checkpoint_key = "_"+k
+            init_args[k]=checkpoint.get(checkpoint_key)
+        init_args["partition_offset"] = partition_offset
+        return StreamingDatasetSplitter(**init_args)
+
+    def get_shard_size(self):
+        return self._shard_size
+
+    def get_default_shard_count(self):
```

Comment:
No code calls the function with the exception of test cases.

Reply:
removed